SalesforceCommerceCloud/pwa-kit
# [Generator] Fix generator failing on Windows using WSL

GUS: https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE00000m7lbMYAQ/view



# Description



The generator fails to create a project on Windows when running the App Generator `npx pwa-kit-create-app` using Windows Subsystem for Linux (WSL) from the Windows drive file system folder mounted in `/mnt/c` or `/mnt/<LOGIC_UNIT>`.

The root of the issue seems to be a lack of permissions on the WSL command line to manipulate the Windows drive temporal folder `/tmp/`. 

The `extractTemplate` function in the App generator tries to move a folder inside the OS temporal directory `/tmp/` to the final directory. The generator process fails due to a lack of permissions.

This PR replaces the `shelljs.mv()` method with the alternative [`shelljs.cp()`](https://github.com/shelljs/shelljs#cpoptions-source_array-dest) method and its `-R` recursively option to copy the template files from the OS temporal director to the final output directory.

----
### Limitations of using WSL to run processes from Windows drive file system folder mounted in `/mnt/c`
The project is now generated successfully, and the project start and works properly when running `npm start`.

However, we start to see other issues related to running processes using WSL from the Windows drive file system folder mounted in `/mnt/c`. 
Like these Webpack watch errors that don't have a proper solution other than not running process from the Windows drive file system when using WSL.

```
adam@WinDev2104Eval:/mnt/c/Users/User/git/pwa-kit/preset-retail-react-app-demo$ npm start

> demo-storefront@0.0.1 start /mnt/c/Users/User/git/pwa-kit/preset-retail-react-app-demo
> cross-env NODE_ICU_DATA=node_modules/full-icu node ./scripts/start ssr

ssr-server: [nodemon] 2.0.14
ssr-server: [nodemon] to restart at any time, enter `rs`
ssr-server: [nodemon] watching path(s): build/build.marker
ssr-server: [nodemon] watching extensions: js,mjs,json
webpack: Watchpack Error (initial scan): Error: EACCES: permission denied, lstat '/mnt/c/DumpStack.log.tmp'
webpack: Watchpack Error (initial scan): Error: EACCES: permission denied, lstat '/mnt/c/pagefile.sys'
webpack: Watchpack Error (initial scan): Error: EACCES: permission denied, lstat '/mnt/c/swapfile.sys'
webpack: Watchpack Error (initial scan): Error: EACCES: permission denied, lstat '/mnt/c/DumpStack.log.tmp'
webpack: Watchpack Error (initial scan): Error: EACCES: permission denied, lstat '/mnt/c/pagefile.sys'
webpack: Watchpack Error (initial scan): Error: EACCES: permission denied, lstat '/mnt/c/swapfile.sys'
webpack: Watchpack Error (initial scan): Error: EACCES: permission denied, lstat '/mnt/c/DumpStack.log.tmp'
webpack: Watchpack Error (initial scan): Error: EACCES: permission denied, lstat '/mnt/c/pagefile.sys'
webpack: Watchpack Error (initial scan): Error: EACCES: permission denied, lstat '/mnt/c/swapfile.sys'
webpack: Watchpack Error (initial scan): Error: EACCES: permission denied, lstat '/mnt/c/DumpStack.log.tmp'
webpack: Watchpack Error (initial scan): Error: EACCES: permission denied, lstat '/mnt/c/pagefile.sys'
webpack: Watchpack Error (initial scan): Error: EACCES: permission denied, lstat '/mnt/c/swapfile.sys'
ssr-server: [nodemon] restarting due to changes...
ssr-server: [nodemon] starting `node /mnt/c/Users/User/git/pwa-kit/preset-retail-react-app-demo-1/build/ssr.js`
ssr-server: Proxying /mobify/proxy/api and /mobify/caching/api to https://kv7kzm78.api.commercecloud.salesforce.com
ssr-server: Proxying /mobify/proxy/ocapi and /mobify/caching/ocapi to https://zzte-053.sandbox.us02.dx.commercecloud.salesforce.com
ssr-server: Proxying /mobify/proxy/einstein and /mobify/caching/einstein to https://api.cquotient.com
ssr-server: HTTP development server listening on http://localhost:3000
```
Related Webpack thread: https://github.com/webpack/watchpack/issues/187#issuecomment-821622126



# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- Use shelljs cp -R instead of mv

# How to Test-Drive This PR

- Enable and install WSL on Windows.
  - Open a command line, choosing Run as administrator.
  - Run the command to enable WSL:
    - `Enable-WindowsOptionalFeature -Online -FeatureName Microsoft-Windows-Subsystem-Linux`
  - Restart Windows.
  - Install a Linux distribution.
    - Launch the Microsoft store.
    - Search for a Linux distribution. e.g. Ubuntu 18.04 LTS
    - Click get.
    - Launch the Linux distribution.
     - Set username and password.
- Access Windows file from the Ubuntu command line by changing the directory: `cd /mnt/c`
- Run the generator and verify that now the project is generated successfully:
```
GENERATOR_PRESET=retail-react-app-demo node packages/pwa-kit-create-app/scripts/create-mobify-app.js --outputDir ./preset-retail-react-app-demo
```

# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
